lib/puppet/pops/model/pn_transformer.rb

```diff
@@ -1,22 +1,18 @@
+require 'puppet/concurrent/thread_local_singleton'
+
 module Puppet::Pops
 module Model
 
 
 class PNTransformer
-  def self.visitor
-    @visitor ||= Visitor.new(nil, 'transform', 0, 0)
-  end
-
-  def self.singleton
-    @singleton ||= new(visitor)
-  end
+  extend Puppet::Concurrent::ThreadLocalSingleton
 
   def self.transform(ast)
     singleton.transform(ast)
   end
 
   def initialize(visitor)
-    @visitor = visitor
+    @visitor = Visitor.new(nil, 'transform', 0, 0)
```

Comment:
Are we intentionally no longer using the `visitor` that gets passed in?

Reply:
good catch, i removed that

Reply:
Are there things calling this that will need to be updated? I'm not familiar with how this is used at all.

Reply:
I think it's just the creation of the singleton, since we try to use that everywhere.